src/Compilers/CSharp/Test/Emit/Emit/DeterministicTests.cs

```diff
@@ -109,6 +110,48 @@ public void Simple()
             Assert.NotEqual(mvid3, mvid7);
         }
 
+        [Fact]
+        public void PlatformChangeGuid()
+        {
+            var source =
+@"class Program
+{
+    public static void Main(string[] args) {}
+}";
+            // Two identical compilations should produce the same MVID
+            var mvid1 = CompiledGuid(source, "X1", false, Platform.X86);
+            var mvid2 = CompiledGuid(source, "X1", false, Platform.X86);
+            Assert.Equal(mvid1, mvid2);
+
+            var mvid3 = CompiledGuid(source, "X1", false, Platform.X64);
+            var mvid4 = CompiledGuid(source, "X1", false, Platform.X64);
+            Assert.Equal(mvid3, mvid4);
+
+            var mvid5 = CompiledGuid(source, "X1", false, Platform.Arm64);
+            var mvid6 = CompiledGuid(source, "X1", false, Platform.Arm64);
+            Assert.Equal(mvid5, mvid6);
+
+            // No two platforms should produce the same MVID
+            Assert.NotEqual(mvid1, mvid3);
```

Comment:
Please test Amd64 vs Arm64.

Reply:
That's done below with the comparison between mvid3 and mvid5

Reply:
I do, see the point where mvid3 and mvid5 are compared. (just below)